### PR TITLE
Translation shawn

### PR DESCRIPTION
**Pull request by ShawnChen1996**

### CODE REVIEW

**Comment by ShawnChen1996** [comment]
There is a conflict but I don' t see why.  @MichaelChirico

**Comment by MichaelChirico** [comment]
@ShawnChen1996 have resolved the conflict, please check:

be4ab0a2d268eb170fd2bd6cc0d1fbf4e62b347f

**Comment by MichaelChirico** [comment]
Thanks Xiao! Please have a look @Xueliang24

**Comment by Xueliang24** [comment]
OK！

Michael Chirico <notifications@github.com> 于2019年11月17日周日 下午8:46写道：

**Comment by Xueliang24** [comment]
I will check it in next week。

Michael Chirico <notifications@github.com> 于2019年11月17日周日 下午8:46写道：

File: po/R-zh_CN.po [hunk 1]
```diff
@@ -1881,101 +1881,106 @@ msgid "but should be a number between 2 and 100"
...
-msgstr ""
+msgstr "key2()已经废弃，请使用indices()"
 
 msgid ""
 "key(x)<-value is deprecated and not supported. Please change to use setkey() "
 "with perhaps copy(). Has been warning since 2012 and will be an error in "
 "future."
-msgstr ""
+msgstr "key(x)<-value 已经废弃，不再支持。请使用setkey(), 也许您需要配套使用copy()"
+"自2012年来就警告将弃用原用法，将来该用法将抛出错误"
```
Comment (Xueliang24) [inline po/R-zh_CN.po]:
```suggestion
"自2012年来就警告将弃用原用法，将来该用法将抛出错误"
```
“将抛出错误”改为“报错”

File: po/R-zh_CN.po [hunk 2]
```diff
@@ -1881,101 +1881,106 @@ msgid "but should be a number between 2 and 100"
...
-msgstr ""
+msgstr "内部错误： 目前在setkey中，'cols'应该是字符类型, 请报告"
 
 msgid "Internal error: index '"
-msgstr ""
+msgstr "内部错误：索引"
 
 msgid "' exists but is invalid"
-msgstr ""
+msgstr "存在但非法"
```
Comment (Xueliang24) [inline po/R-zh_CN.po]: invalid无效

File: po/R-zh_CN.po [hunk 3]
```diff
@@ -1881,101 +1881,106 @@ msgid "but should be a number between 2 and 100"
...
 
 msgid ""
 "Use 'if (length(o <- forderv(DT,by))) ...' for efficiency in one step, so "
 "you have o as well if not sorted."
 msgstr ""
+"请使用'if (length(o <- forderv(DT,by))) ...' , 以便在一步中拥有较好的效率，"
+"同时如果你还未排序，你也获得了变量o"
```
Comment (Xueliang24) [inline po/R-zh_CN.po]:
```suggestion
"同时如果你还未排序，你也获得了变量o"
```你也有变量0

Comment (Xueliang24) [inline po/R-zh_CN.po]:
```suggestion
"同时如果你还未排序，你也获得了变量o"
```
“所以如果你还未排序，你也有变量0”

File: po/R-zh_CN.po [hunk 4]
```diff
@@ -1881,101 +1881,106 @@ msgid "but should be a number between 2 and 100"
...
 
 msgid "x is vector but 'by' is supplied"
-msgstr ""
+msgstr "x是一个向量， 但是'by'提供了"
```
Comment (Xueliang24) [inline po/R-zh_CN.po]:
```suggestion
msgstr "x是一个向量， 但是'by'提供了"
```
“x是一个向量， 但是提供了'by'”

File: po/zh_CN.po [hunk 5]
```diff
@@ -3726,213 +3726,227 @@ msgstr ""
...
-msgstr ""
+msgstr "输入项 %d 不是一个data.frame, data.table 或list"
 
 #: rbindlist.c:39
 #, c-format
 msgid ""
 "Item %d has %d columns, inconsistent with item %d which has %d columns. To "
 "fill missing columns use fill=TRUE."
 msgstr ""
+"项%d有%d列, 与第%d项不一致， 它有%d列. 若要填补这些列, 请使用fill=TRUE."
```
Comment (Xueliang24) [inline po/zh_CN.po]:
```suggestion
"第%d项有%d列, 与第%d项不一致， 它有%d列. 若要填补这些列, 请使用fill=TRUE."
```

File: po/zh_CN.po [hunk 6]
```diff
@@ -3726,213 +3726,227 @@ msgstr ""
...
 
 #: init.c:248
 msgid "The ascii character ':' is not just after '9'"
-msgstr ""
+msgstr "ascii字符':'不是在'9'后"
 
 #: init.c:249
 msgid "The C expression (uint_fast8_t)('9'-':')<10 is true. Should be false."
-msgstr ""
+msgstr "C表达式(uint_fast8_t)('9'-':')<10 <10  为 true. 应该是 false."
```
Comment (Xueliang24) [inline po/zh_CN.po]:
```suggestion
msgstr "C表达式(uint_fast8_t)('9'-':')<10  为 true. 应该是 false."
```

File: po/R-zh_CN.po (hunk 3 above)
Comment (ShawnChen1996) [reply po/R-zh_CN.po]: I think "获得" is a better word here, cuz it explains if you have not sorted it before, you can obtain the variable `o` by this code. "有" sounds stiff here

**Comment by MichaelChirico** [comment]
Hey @ShawnChen1996 thanks again for your work! Please have a look at Xueliang's suggestions when you get a chance

**Comment by ShawnChen1996** [comment]
I have already revised the translation https://github.com/Rdatatable/data.table/pull/4050/commits/e26b10ce02a7e222fff6305225cb01f18a76b3b0 according to his suggestions except for one place https://github.com/Rdatatable/data.table/pull/4050#discussion_r348520787. And I have marked all other suggestion as resolved.

**Comment by MichaelChirico** [comment]
OK thanks for confirming! Merging. Thanks @ShawnChen1996 and @Xueliang24!!

**Pull request merged**